src/model/immutable/ContentState.js

```diff
@@ -56,6 +55,17 @@ class ContentState extends ContentStateRecord {
     return block;
   }
 
+  getFirstLevelBlocks(): BlockMap {
+    return this.getBlockChildren('');
+  }
+
+  getBlockChildren(key: string): BlockMap {
```

Comment:
This method is O(N), and by being used in `Component#shouldComponentUpdate()` leads to O(N²) complexity which makes the editor quickly unusable (e.g. try with 1000 blocks — `printf 'text\n%.0s' {1..1000} | pbcopy`).